pkg/machine/ignition/ignition.go

```diff
@@ -668,6 +684,24 @@ Delegate=memory pids cpu io
 		})
 	}
 
+	RosettaActivation := `#!/bin/bash
+echo ":rosetta:M::\x7fELF\x02\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x00\x3e\x00:\xff\xff\xff\xff\xff\xfe\xfe\x00\xff\xff\xff\xff\xff\xff\xff\xff\xfe\xff\xff\xff:/mnt/rosetta:F" > /proc/sys/fs/binfmt_misc/register
```

Comment:
if this is a one liner anyway then why not just directly write it to the unit file.

Reply:
Writing this in the unit file will work.
```
	rosettaUnit.Add("Service", "ExecStart",`/bin/sh -c "echo ':rosetta:M::\\x7fELF\\x02\\x01\\x01\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x02\\x00\\x3e\\x00:\\xff\\xff\\xff\\xff\\xff\\xfe\\xfe\\x00\\xff\\xff\\xff\\xff\\xff\\xff\\xff\\xff\\xfe\\xff\\xff\\xff:/mnt/rosetta:F' > /proc/sys/fs/binfmt_misc/register"`)
```
However, it requires additional escaping and is dirtier than the Apple's sample code.
https://developer.apple.com/documentation/virtualization/running_intel_binaries_in_linux_vms_with_rosetta

Which is better?

Reply:
I think one unit file is better but don't feel strongly about it